### ISSUE TITLE
Example not working?

### ISSUE THREAD

**Issue by eplondke** [mvitez/thnets#5]
I'm trying to follow the directions for the example, but am perhaps doing something wrong.  Perhaps I need a certain image size?

I downloaded the demo from https://www.dropbox.com/sh/u3bunkfm0dzjix6/AABQ4Nq4-70MU57MxXyyjrrMa?dl=0

I compiled default on x86.

I get the assertion: Given input size: (256x2x4). Calculated output size: (4096x-3x-1). Output size is too small

./test -m default_model -i ~/panda.jpg -d 2
nn module nn.Sequential with 4 elements
1/4: "gradInput"
    Float tensor of dimension 0
2/4: "modules"
    Table long 20
    1/20: 1.000000
        nn module nn.SpatialConvolutionMM with 17 elements
        1/17: "padW"
            2.000000
        2/17: "nInputPlane"
            3.000000
        3/17: "output"
            Float tensor of dimension 0
        4/17: "gradInput"
            Float tensor of dimension 0
        5/17: "_type"
            "torch.FloatTensor"
        6/17: "dH"
            4.000000
        7/17: "dW"
            4.000000
        8/17: "nOutputPlane"
            64.000000
        9/17: "bias"
            Float tensor of dimension 1 (64)
            offset = 0, strides = (1)
        10/17: "kH"
            11.000000
        11/17: "finput"
            Float tensor of dimension 0
        12/17: "weight"
            Float tensor of dimension 2 (64,363)
            offset = 0, strides = (363,1)
        13/17: "gradWeight"
            Float tensor of dimension 2 (64,363)
            offset = 0, strides = (363,1)
        14/17: "gradBias"
            Float tensor of dimension 1 (64)
            offset = 0, strides = (1)
        15/17: "padH"
            2.000000
        16/17: "fgradInput"
            Float tensor of dimension 0
        17/17: "kW"
            11.000000
    2/20: 2.000000
        nn module nn.ReLU with 6 elements
        1/6: "inplace"
            true
        2/6: "threshold"
            0.000000
        3/6: "_type"
            "torch.FloatTensor"
        4/6: "output"
            Float tensor of dimension 0
        5/6: "gradInput"
            Float tensor of dimension 0
        6/6: "val"
            0.000000
    3/20: 3.000000
        nn module nn.SpatialMaxPooling with 11 elements
        1/11: "dH"
            2.000000
        2/11: "dW"
            2.000000
        3/11: "kW"
            3.000000
        4/11: "gradInput"
            Float tensor of dimension 0
        5/11: "indices"
            Float tensor of dimension 0
        6/11: "_type"
            "torch.FloatTensor"
        7/11: "padW"
            0.000000
        8/11: "ceil_mode"
            false
        9/11: "kH"
            3.000000
        10/11: "output"
            Float tensor of dimension 0
        11/11: "padH"
            0.000000
    4/20: 4.000000
        nn module nn.SpatialConvolutionMM with 17 elements
        1/17: "padW"
            2.000000
        2/17: "nInputPlane"
            64.000000
        3/17: "output"
            Float tensor of dimension 0
        4/17: "gradInput"
            Float tensor of dimension 0
        5/17: "_type"
            "torch.FloatTensor"
        6/17: "dH"
            1.000000
        7/17: "dW"
            1.000000
        8/17: "nOutputPlane"
            192.000000
        9/17: "bias"
            Float tensor of dimension 1 (192)
            offset = 0, strides = (1)
        10/17: "kH"
            5.000000
        11/17: "finput"
            Float tensor of dimension 0
        12/17: "weight"
            Float tensor of dimension 2 (192,1600)
            offset = 0, strides = (1600,1)
        13/17: "gradWeight"
            Float tensor of dimension 2 (192,1600)
            offset = 0, strides = (1600,1)
        14/17: "gradBias"
            Float tensor of dimension 1 (192)
            offset = 0, strides = (1)
        15/17: "padH"
            2.000000
        16/17: "fgradInput"
            Float tensor of dimension 0
        17/17: "kW"
            5.000000
    5/20: 5.000000
        nn module nn.ReLU with 6 elements
        1/6: "inplace"
            true
        2/6: "threshold"
            0.000000
        3/6: "_type"
            "torch.FloatTensor"
        4/6: "output"
            Float tensor of dimension 0
        5/6: "gradInput"
            Float tensor of dimension 0
        6/6: "val"
            0.000000
    6/20: 6.000000
        nn module nn.SpatialMaxPooling with 11 elements
        1/11: "dH"
            2.000000
        2/11: "dW"
            2.000000
        3/11: "kW"
            3.000000
        4/11: "gradInput"
            Float tensor of dimension 0
        5/11: "indices"
            Float tensor of dimension 0
        6/11: "_type"
            "torch.FloatTensor"
        7/11: "padW"
            0.000000
        8/11: "ceil_mode"
            false
        9/11: "kH"
            3.000000
        10/11: "output"
            Float tensor of dimension 0
        11/11: "padH"
            0.000000
    7/20: 7.000000
        nn module nn.SpatialConvolutionMM with 17 elements
        1/17: "padW"
            1.000000
        2/17: "nInputPlane"
            192.000000
        3/17: "output"
            Float tensor of dimension 0
        4/17: "gradInput"
            Float tensor of dimension 0
        5/17: "_type"
            "torch.FloatTensor"
        6/17: "dH"
            1.000000
        7/17: "dW"
            1.000000
        8/17: "nOutputPlane"
            384.000000
        9/17: "bias"
            Float tensor of dimension 1 (384)
            offset = 0, strides = (1)
        10/17: "kH"
            3.000000
        11/17: "finput"
            Float tensor of dimension 0
        12/17: "weight"
            Float tensor of dimension 2 (384,1728)
            offset = 0, strides = (1728,1)
        13/17: "gradWeight"
            Float tensor of dimension 2 (384,1728)
            offset = 0, strides = (1728,1)
        14/17: "gradBias"
            Float tensor of dimension 1 (384)
            offset = 0, strides = (1)
        15/17: "padH"
            1.000000
        16/17: "fgradInput"
            Float tensor of dimension 0
        17/17: "kW"
            3.000000
    8/20: 8.000000
        nn module nn.ReLU with 6 elements
        1/6: "inplace"
            true
        2/6: "threshold"
            0.000000
        3/6: "_type"
            "torch.FloatTensor"
        4/6: "output"
            Float tensor of dimension 0
        5/6: "gradInput"
            Float tensor of dimension 0
        6/6: "val"
            0.000000
    9/20: 9.000000
        nn module nn.SpatialConvolutionMM with 17 elements
        1/17: "padW"
            1.000000
        2/17: "nInputPlane"
            384.000000
        3/17: "output"
            Float tensor of dimension 0
        4/17: "gradInput"
            Float tensor of dimension 0
        5/17: "_type"
            "torch.FloatTensor"
        6/17: "dH"
            1.000000
        7/17: "dW"
            1.000000
        8/17: "nOutputPlane"
            256.000000
        9/17: "bias"
            Float tensor of dimension 1 (256)
            offset = 0, strides = (1)
        10/17: "kH"
            3.000000
        11/17: "finput"
            Float tensor of dimension 0
        12/17: "weight"
            Float tensor of dimension 2 (256,3456)
            offset = 0, strides = (3456,1)
        13/17: "gradWeight"
            Float tensor of dimension 2 (256,3456)
            offset = 0, strides = (3456,1)
        14/17: "gradBias"
            Float tensor of dimension 1 (256)
            offset = 0, strides = (1)
        15/17: "padH"
            1.000000
        16/17: "fgradInput"
            Float tensor of dimension 0
        17/17: "kW"
            3.000000
    10/20: 10.000000
        nn module nn.ReLU with 6 elements
        1/6: "inplace"
            true
        2/6: "threshold"
            0.000000
        3/6: "_type"
            "torch.FloatTensor"
        4/6: "output"
            Float tensor of dimension 0
        5/6: "gradInput"
            Float tensor of dimension 0
        6/6: "val"
            0.000000
    11/20: 11.000000
        nn module nn.SpatialConvolutionMM with 17 elements
        1/17: "padW"
            1.000000
        2/17: "nInputPlane"
            256.000000
        3/17: "output"
            Float tensor of dimension 0
        4/17: "gradInput"
            Float tensor of dimension 0
        5/17: "_type"
            "torch.FloatTensor"
        6/17: "dH"
            1.000000
        7/17: "dW"
            1.000000
        8/17: "nOutputPlane"
            256.000000
        9/17: "bias"
            Float tensor of dimension 1 (256)
            offset = 0, strides = (1)
        10/17: "kH"
            3.000000
        11/17: "finput"
            Float tensor of dimension 0
        12/17: "weight"
            Float tensor of dimension 2 (256,2304)
            offset = 0, strides = (2304,1)
        13/17: "gradWeight"
            Float tensor of dimension 2 (256,2304)
            offset = 0, strides = (2304,1)
        14/17: "gradBias"
            Float tensor of dimension 1 (256)
            offset = 0, strides = (1)
        15/17: "padH"
            1.000000
        16/17: "fgradInput"
            Float tensor of dimension 0
        17/17: "kW"
            3.000000
    12/20: 12.000000
        nn module nn.ReLU with 6 elements
        1/6: "inplace"
            true
        2/6: "threshold"
            0.000000
        3/6: "_type"
            "torch.FloatTensor"
        4/6: "output"
            Float tensor of dimension 0
        5/6: "gradInput"
            Float tensor of dimension 0
        6/6: "val"
            0.000000
    13/20: 13.000000
        nn module nn.SpatialMaxPooling with 11 elements
        1/11: "dH"
            2.000000
        2/11: "dW"
            2.000000
        3/11: "kW"
            3.000000
        4/11: "gradInput"
            Float tensor of dimension 0
        5/11: "indices"
            Float tensor of dimension 0
        6/11: "_type"
            "torch.FloatTensor"
        7/11: "padW"
            0.000000
        8/11: "ceil_mode"
            false
        9/11: "kH"
            3.000000
        10/11: "output"
            Float tensor of dimension 0
        11/11: "padH"
            0.000000
    14/20: 14.000000
        nn module nn.View with 6 elements
        1/6: "train"
            false
        2/6: "output"
            Float tensor of dimension 0
        3/6: "gradInput"
            Float tensor of dimension 0
        4/6: "numElements"
            9216.000000
        5/6: "size"
            Long storage of 1 elements
        6/6: "_type"
            "torch.FloatTensor"
    15/20: 15.000000
        nn module nn.Linear with 9 elements
        1/9: "gradBias"
            Float tensor of dimension 1 (4096)
            offset = 40220736, strides = (1)
        2/9: "bias"
            Float tensor of dimension 1 (4096)
            offset = 40220736, strides = (1)
        3/9: "gradInput"
            Float tensor of dimension 0
        4/9: "weight"
            Float tensor of dimension 2 (4096,9216)
            offset = 2472000, strides = (9216,1)
        5/9: "train"
            false
        6/9: "gradWeight"
            Float tensor of dimension 2 (4096,9216)
            offset = 2472000, strides = (9216,1)
        7/9: "addBuffer"
            Float tensor of dimension 1 (512)
            offset = 0, strides = (1)
        8/9: "output"
            Float tensor of dimension 0
        9/9: "_type"
            "torch.FloatTensor"
    16/20: 16.000000
        nn module nn.ReLU with 7 elements
        1/7: "inplace"
            false
        2/7: "threshold"
            0.000000
        3/7: "val"
            0.000000
        4/7: "output"
            Float tensor of dimension 0
        5/7: "gradInput"
            Float tensor of dimension 0
        6/7: "train"
            false
        7/7: "_type"
            "torch.FloatTensor"
    17/20: 17.000000
        nn module nn.Linear with 9 elements
        1/9: "gradBias"
            Float tensor of dimension 1 (4096)
            offset = 57002048, strides = (1)
        2/9: "bias"
            Float tensor of dimension 1 (4096)
            offset = 57002048, strides = (1)
        3/9: "gradInput"
            Float tensor of dimension 0
        4/9: "weight"
            Float tensor of dimension 2 (4096,4096)
            offset = 40224832, strides = (4096,1)
        5/9: "train"
            false
        6/9: "gradWeight"
            Float tensor of dimension 2 (4096,4096)
            offset = 40224832, strides = (4096,1)
        7/9: "addBuffer"
            Float tensor of dimension 1 (512)
            offset = 0, strides = (1)
        8/9: "output"
            Float tensor of dimension 0
        9/9: "_type"
            "torch.FloatTensor"
    18/20: 18.000000
        nn module nn.ReLU with 7 elements
        1/7: "inplace"
            false
        2/7: "threshold"
            0.000000
        3/7: "val"
            0.000000
        4/7: "output"
            Float tensor of dimension 0
        5/7: "gradInput"
            Float tensor of dimension 0
        6/7: "train"
            false
        7/7: "_type"
            "torch.FloatTensor"
    19/20: 19.000000
        nn module nn.Linear with 9 elements
        1/9: "gradBias"
            Float tensor of dimension 1 (5000)
            offset = 77486144, strides = (1)
        2/9: "bias"
            Float tensor of dimension 1 (5000)
            offset = 77486144, strides = (1)
        3/9: "gradInput"
            Float tensor of dimension 0
        4/9: "weight"
            Float tensor of dimension 2 (5000,4096)
            offset = 57006144, strides = (4096,1)
        5/9: "train"
            false
        6/9: "gradWeight"
            Float tensor of dimension 2 (5000,4096)
            offset = 57006144, strides = (4096,1)
        7/9: "addBuffer"
            Float tensor of dimension 1 (512)
            offset = 0, strides = (1)
        8/9: "output"
            Float tensor of dimension 0
        9/9: "_type"
            "torch.FloatTensor"
    20/20: 20.000000
        nn module nn.SoftMax with 3 elements
        1/3: "gradInput"
            Float tensor of dimension 0
        2/3: "_type"
            "torch.FloatTensor"
        3/3: "output"
            Float tensor of dimension 0
3/4: "_type"
    "torch.FloatTensor"
4/4: "output"
    Float tensor of dimension 0
1) 2 3 64 29 44 44
2) 7 3 64 29 44 44
3) 4 3 64 14 21 21
4) 2 3 192 14 21 21
5) 7 3 192 14 21 21
6) 4 3 192 6 10 10
7) 2 3 384 6 10 10
8) 7 3 384 6 10 10
9) 2 3 256 6 10 10
10) 7 3 256 6 10 10
11) 2 3 256 6 10 10
12) 7 3 256 6 10 10
13) 4 3 256 2 4 4
Given input size: (256x2x4). Calculated output size: (4096x-3x-1). Output size is too small

**Comment by eplondke** [comment]
Oops! My image size was invalid (too small).
A different image runs to completion.

**Comment by mvitez** [comment]
Excellent. Yes, the image size has to be at least 224x224 or something around there.
